src/variant_iter.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::iter::{ExactSizeIterator, Iterator};
+
+use variant::Variant;
+
```

Comment:
Maybe add a small comment here why we don't just add bindings for `GVariantIter` but do our own simple version of it